tests/unit/Mage/Api/Model/UserTest.php

```diff
@@ -0,0 +1,173 @@
+<?php
+
+/**
+ * OpenMage
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available at https://opensource.org/license/osl-3-0-php
+ *
+ * @category   OpenMage
+ * @package    OpenMage_Tests
+ * @copyright  Copyright (c) 2025 The OpenMage Contributors (https://www.openmage.org)
+ * @license    https://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace OpenMage\Tests\Unit\Mage\Api\Model;
+
+use Generator;
+use Mage;
+use Mage_Api_Model_User as Subject;
+use PHPUnit\Framework\TestCase;
+
+class UserTest extends TestCase
+{
+    /** @phpstan-ignore property.onlyWritten */
+    private static Subject $subject;
+
+    public static function setUpBeforeClass(): void
+    {
+        Mage::app();
+        self::$subject = Mage::getModel('api/user');
+    }
+
+    /**
+     * @dataProvider provideValidateApiUserData
+     * @param array|true $expectedResult
+     * @group Mage_Api
+     * @group Mage_Api_Model
+     */
+    public function testValidate($expectedResult, array $methods): void
+    {
+        $mock = $this->getMockBuilder(Subject::class)
+            ->setMethods([
```

Comment:
If the `$methods` array is controlled by the dataProvider,
why not loop the data there?

`->setMethods(array_keys($methods))`
 and
```php
foreach ($methods as $key => $value) {
  $mock->method($key)->willReturn($value);
}
```

with that, you might be able to reduce the amount of duplicated lines